src/TRestAxionOpticsMirror.cxx

```diff
@@ -0,0 +1,630 @@
+    cout << "Downloading from Henke database. This process will take a long time." << endl;
+    cout << "After producing these files, please, contribute them to the " << endl;
+    cout << "rest-for-physics/axionlib-data inside the optics directory ..." << endl;
+    cout << "---------------------------------- INFO ------------------------------" << endl;
+
+    fReflectivityTable.clear();
+    fTransmissionTable.clear();
+    map<Int_t, std::vector<Float_t>> reflectivity;
+    map<Int_t, std::vector<Float_t>> transmission;
+    for (double n = 0; n < 9; n = n + 4.5) {
+        fHenkeKeys["Min"] = DoubleToString(n);
+        fHenkeKeys["Max"] = DoubleToString(n + 4.5);
+
+        cout.flush();
+        vector<Float_t> reflect;
+        vector<Float_t> transm;
+        reflect.clear();
+        transm.clear();
+
+        cout << "Scanning angles between " << n << " and " << n + 4.5 << ".";
+        for (int e = 30; e <= 15000; e += 30) {
+            fHenkeKeys["Fixed"] = IntegerToString(e);
+            cout << ".";
+            cout.flush();
+            string fname = DownloadHenkeFile();
+
+            std::vector<std::vector<Double_t>> data;
+            TRestTools::ReadASCIITable(fname, data, 2);
+
+            // we skip the last point if we are not at the latest angles file
+            Int_t N = data.size() - 1;
+            if (n == 4.5) N = N + 1;
+
+            for (int m = 0; m < N; m++) reflectivity[e].push_back((Float_t)data[m][1]);
+            for (int m = 0; m < N; m++) transmission[e].push_back((Float_t)data[m][2]);
+        }
+        cout << endl;
+    }
+    cout << "Number of angles stored : " << reflectivity[30].size() << endl;
+
+    for (const auto& x : reflectivity) {
+        fReflectivityTable.push_back(x.second);
+    }
+    for (const auto& x : transmission) {
+        fTransmissionTable.push_back(x.second);
+    }
+
+    ExportTables();
+}
+
+///////////////////////////////////////////////
+/// \brief It is a private method to export the tables to a binary file once the tables
+/// have been downloaded from Henke database
+///
+Int_t TRestAxionOpticsMirror::ExportTables() {
+    if (fReflectivityTable.size() == 0) {
+        ferr << "Nothing to export!" << endl;
+        return 1;
+    }
+
+    string path = REST_USER_PATH + "/export/";
+
+    if (!TRestTools::fileExists(path)) {
+        cout << "Creating path: " << path << endl;
+        system(("mkdir -p " + path).c_str());
+    }
+
+    string fnameR = fMirrorType + "_" + fLayer + "_" + fLayerThickness + "_" + fSubstrate + "_" + fSigma1 +
+                    ".reflectivity";
+    TRestTools::ExportBinaryTable(path + fnameR, fReflectivityTable);
+
+    info << "Reflectivity table generated at: " << path + fnameR << endl;
+
+    string fnameT = fMirrorType + "_" + fLayer + "_" + fLayerThickness + "_" + fSubstrate + "_" + fSigma1 +
+                    ".transmission";
+    TRestTools::ExportBinaryTable(path + fnameT, fTransmissionTable);
```

Comment:
Given that these are stored in raw binary files, I would propose to change the naming scheme according to the ideas presented here:

https://github.com/c-blake/nio

So essentially the file name extension becomes a description of the data layout + type. That makes in unambiguous what's stored in the file & easy to write a custom parser for it in case one wants to look at the data elsewhere or just at a later time quickly understand what the data in the file is stored like.

So for example for the reflectivity file, if we have 500 energies that we store at 901 angles (this is from an example Johanna sent me), the file name would become:
```
<foo>.N901f
```
where `N` denotes we have `N` rows (determined based on file size + the rest of the extension), `901` elements per row and `f` denoting `float` in C typical notation for a 4 byte floating point number. 

That way one can easily write a safe, runtime or compile time based parser that only needs to look at the file size and the extension to create a (possibly nested) container to store the data in the correct type and order.

Reply:
Updated at dd96cab